keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,842 @@
+- If deletionPolicy is Retain, then Bucket is left as is, but the BucketRequest is deleted.
+
+Only when all accessors (BucketClaims) of the Bucket are deleted, is the Bucket itself cleaned up. An orphaned Bucket can be "recovered"  by manually creating a new BucketRequest object that points to the orphaned Bucket. This manual style of associating BucketRequests to Buckets will only succeed if the Bucket is not already bound to a different BucketRequest.
 
-message ProvisionerGetInfoResponse {
-  // The name MUST follow domain name notation format
-  // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
-  // include the plugin's host company name and the plugin name,
-  // to minimize the possibility of collisions. It MUST be 63
-  // characters or less, beginning and ending with an alphanumeric
-  // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
-  // alphanumerics between. This field is REQUIRED.
-  string name = 1;
+When a user deletes a BucketClaim, the corresponding secret/serviceaccount are also deleted. If a pod has that secret mounted when delete is called, then the secret will not be deleted, but will instead have its deletionTimestamp set. In this way, access to a Bucket is preserved until the application pod dies. 
+
+When an admin deletes any of the Class objects, it does not affect existing Buckets as they store a copy of the fields in the class object. If a Bucket  is manually deleted by an admin, then the cluster is considered to be in an invalid state. Manual recovery is possible if data is not already lost.  
+
+# COSI API Reference
+
+## Bucket
+
+Resource to represent a Bucket in OSP. Buckets are cluster-scoped.
+
+```yaml
+Bucket {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketSpec {
+    // Provisioner is the name of driver associated with this bucket
+    Provisioner string
+
+    // DeletionPolicy is used to specify how COSI should handle deletion of this
+    // bucket. There are 3 possible values:
+    //  - Retain: Indicates that the bucket should not be deleted from the OSP
+    //  - Delete: Indicates that the bucket should be deleted from the OSP
+    //        once all the workloads accessing this bucket are done
+    //  - ForceDelete: Indicates that the bucket and its contents should be
+    //        deleted from the OSP even if workloads are currently accessing
+    //        this bucket
+    DeletionPolicy DeletionPolicy
+
+    // Name of the BucketClass specified in the BucketRequest
+    // +optional
+    BucketClassName  string
+
+    // Name of the BucketClaim that resulted in the creation of this Bucket
+    // +optional
+    BucketClaim corev1.ObjectReference
+
+    // Protocol is the data API this bucket is expected to adhere to.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocol Protocol
```

Comment:
I don't see a specific definition for the `Protocol` type. Is that intentional, or is it defined elsewhere?

Reply:
Protocol is just a typed string that can be one of 3 values: GCS, AZURE, AWS.